src/sonic-bgpcfgd/tests/test_allow_list.py

```diff
@@ -0,0 +1,296 @@
+        'bgp community-list standard COMMUNITY_ALLOW_LIST_DEPLOYMENT_ID_5_COMMUNITY_1010:2020 permit 1010:2020',
+        'route-map ALLOW_LIST_DEPLOYMENT_ID_5_V4 permit 10',
+        ' match ip address prefix-list PL_ALLOW_LIST_DEPLOYMENT_ID_5_COMMUNITY_1010:2020_V4',
+        ' match community COMMUNITY_ALLOW_LIST_DEPLOYMENT_ID_5_COMMUNITY_1010:2020',
+        'route-map ALLOW_LIST_DEPLOYMENT_ID_5_V6 permit 10',
+        ' match ipv6 address prefix-list PL_ALLOW_LIST_DEPLOYMENT_ID_5_COMMUNITY_1010:2020_V6',
+        ' match community COMMUNITY_ALLOW_LIST_DEPLOYMENT_ID_5_COMMUNITY_1010:2020',
+        ""
+    ]
+    common_objs = {
+        'directory': Directory(),
+        'cfg_mgr':   cfg_mgr,
+        'tf':        TemplateFabric(),
+        'constants': global_constants,
+    }
+    mgr = BGPAllowListMgr(common_objs, "CONFIG_DB", "BGP_ALLOWED_PREFIXES")
+    mgr.set_handler("DEPLOYMENT_ID|5|1010:2020", {
+        "prefixes_v4": "10.20.30.0/24,30.50.0.0/16",
+        "prefixes_v6": "fc00:20::/64,fc00:30::/64",
+    })
+    assert not cfg_mgr.push_list.called, "cfg_mgr.push_list was called, but it shouldn't have been"
+
+#@pytest.mark.skip()
+def test_set_handler_no_community_data_is_already_presented():
+    cfg_mgr = MagicMock()
+    cfg_mgr.update.return_value = None
+    cfg_mgr.get_text.return_value = [
+        'ip prefix-list PL_ALLOW_LIST_DEPLOYMENT_ID_5_COMMUNITY_empty_V4 seq 10 deny 0.0.0.0/0 le 17',
+        'ip prefix-list PL_ALLOW_LIST_DEPLOYMENT_ID_5_COMMUNITY_empty_V4 seq 20 permit 20.20.30.0/24 ge 25',
+        'ip prefix-list PL_ALLOW_LIST_DEPLOYMENT_ID_5_COMMUNITY_empty_V4 seq 30 permit 40.50.0.0/16 ge 17',
+        'ipv6 prefix-list PL_ALLOW_LIST_DEPLOYMENT_ID_5_COMMUNITY_empty_V6 seq 10 deny 0::/0 le 59',
+        'ipv6 prefix-list PL_ALLOW_LIST_DEPLOYMENT_ID_5_COMMUNITY_empty_V6 seq 20 deny 0::/0 ge 65',
+        'ipv6 prefix-list PL_ALLOW_LIST_DEPLOYMENT_ID_5_COMMUNITY_empty_V6 seq 30 permit fc01:20::/64 ge 65',
+        'ipv6 prefix-list PL_ALLOW_LIST_DEPLOYMENT_ID_5_COMMUNITY_empty_V6 seq 40 permit fc01:30::/64 ge 65',
+        'route-map ALLOW_LIST_DEPLOYMENT_ID_5_V4 permit 30000',
+        ' match ip address prefix-list PL_ALLOW_LIST_DEPLOYMENT_ID_5_COMMUNITY_empty_V4',
+        'route-map ALLOW_LIST_DEPLOYMENT_ID_5_V6 permit 30000',
+        ' match ipv6 address prefix-list PL_ALLOW_LIST_DEPLOYMENT_ID_5_COMMUNITY_empty_V6',
+        'exit',
+        ""
+    ]
+    common_objs = {
+            'directory': Directory(),
+            'cfg_mgr': cfg_mgr,
+            'tf': TemplateFabric(),
+            'constants': global_constants,
+    }
+    mgr = BGPAllowListMgr(common_objs, "CONFIG_DB", "BGP_ALLOWED_PREFIXES")
+    mgr.set_handler("DEPLOYMENT_ID|5", {
+        "prefixes_v4": "20.20.30.0/24,40.50.0.0/16",
+        "prefixes_v6": "fc01:20::/64,fc01:30::/64",
+    })
+    assert not cfg_mgr.push_list.called, "cfg_mgr.push_list was called, but it shouldn't have been"
```

Comment:
the input seems good, why push_list should not be called in this case?

Reply:
Because the data is already presented in FRR and we don't need to push it again into FRR. I made the test to make sure that all allow_list matching logic works correct.